qualification/__init__.py

```diff
@@ -576,21 +581,27 @@ def create_tied_array_channelised_voltage_receive_stream(
     n_beams = len(multicast_endpoints)
     expected_payload_size = n_chans_per_substream * n_spectra_per_heap * COMPLEX * n_bits_per_sample // 8
     timestamp_step = n_spectra_per_heap * n_samples_between_spectra
+    beam_ants_dtype = np.dtype(np.uint16)
+    beam_ants_itemsize = beam_ants_dtype.itemsize
 
     @numba.cfunc(types.void(types.CPointer(chunk_place_data), types.size_t, types.CPointer(types.int64)), nopython=True)
     def chunk_place(data_ptr, data_size, user_data_ptr):
         data = numba.carray(data_ptr, 1)
         user_data = numba.carray(user_data_ptr, 1)  # Contains the beam index
-        items = numba.carray(intp_to_voidptr(data[0].items), 3, dtype=np.int64)
+        items = numba.carray(intp_to_voidptr(data[0].items), 4, dtype=np.int64)
+        extra = numba.carray(intp_to_voidptr(data[0].extra), 1, dtype=beam_ants_dtype)
         channel_offset = items[0]
         timestamp = items[1]
-        payload_size = items[2]
+        payload_size = items[3]
         beam = user_data[0]
         # If the payload size doesn't match, discard the heap (could be descriptors etc).
         if payload_size == expected_payload_size and timestamp >= 0:
             data[0].chunk_id = timestamp // timestamp_step
             data[0].heap_index = channel_offset // n_chans_per_substream + beam * n_substreams
             data[0].heap_offset = data[0].heap_index * payload_size
+            extra[0] = items[2]  # beam_ants
+            data[0].extra_size = beam_ants_itemsize
+            data[0].extra_offset = data[0].heap_index * beam_ants_itemsize
```

Comment:
I see this follows the format in `src/.../fgpu/recv.py`, which is great. However, entirely unrelated, regarding "accommodating for `extra` item(s)"
* Would you be open to e.g. me writing some kind of "How to write a `chunk_place` callback"?
* Or is there already some note on this in your recent tutorial update?

Reply:
It's documented at https://spead2.readthedocs.io/en/latest/recv-chunk.html#extra-data. I don't think it's in the tutorials yet, but it might go into a later round once the current set have been reviewed.